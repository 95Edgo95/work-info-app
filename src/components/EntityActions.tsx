import Icon from "components/Icon";
import * as React from "react";

import * as styles from "assets/css/entityActions.css";

export default function EntityActions({ handleEdit, handleDelete }) {
  return (
    <div className={styles.actions}>
      <span
        className={styles.edit}
        onClick={handleEdit}
      >
        <Icon name="pencil" color="inherit" size={16} />
      </span>
      <span
        className={styles.delete}
        onClick={handleDelete}
      >
        <Icon name="trash" color="inherit" size={16} />
      </span>
    </div>
  );
}
