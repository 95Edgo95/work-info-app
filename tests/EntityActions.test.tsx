import EntityActions from "../src/components/EntityActions";
import { mount } from "enzyme";
import * as React from "react";

const clickFn = jest.fn();

describe('EntityActions', () => {
  it('should render correctly', () => {
    const component = mount(
      <EntityActions
        handleDelete={clickFn}
        handleEdit={clickFn}
      />
    );

    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
