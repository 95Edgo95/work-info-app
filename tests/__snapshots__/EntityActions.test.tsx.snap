// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityActions should render correctly 1`] = `
<EntityActions
  handleDelete={[MockFunction]}
  handleEdit={[MockFunction]}
>
  <div>
    <span
      onClick={[MockFunction]}
    >
      <Icon
        color="inherit"
        name="pencil"
        size={16}
      >
        <span
          className="undefined "
          role="presentation"
          style={
            Object {
              "color": "inherit",
              "fontSize": "16px",
            }
          }
        >
          <svg>
            <use
              xlinkHref="#pencil"
            />
          </svg>
        </span>
      </Icon>
    </span>
    <span
      onClick={[MockFunction]}
    >
      <Icon
        color="inherit"
        name="trash"
        size={16}
      >
        <span
          className="undefined "
          role="presentation"
          style={
            Object {
              "color": "inherit",
              "fontSize": "16px",
            }
          }
        >
          <svg>
            <use
              xlinkHref="#trash"
            />
          </svg>
        </span>
      </Icon>
    </span>
  </div>
</EntityActions>
`;
